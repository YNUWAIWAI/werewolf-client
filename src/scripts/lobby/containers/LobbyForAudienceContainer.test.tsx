import * as ActionTypes from '../constants/ActionTypes'
import * as React from 'react'
import IntlProviderContainer from './IntlProviderContainer'
import LobbyForAudience from '../components/templates/LobbyForAudience'
import LobbyForAudienceContainer from './LobbyForAudienceContainer'
import {Provider} from 'react-redux'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

describe('<LobbyForAudienceContainer />', () => {
  test('render', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <LobbyForAudienceContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('transition', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <LobbyForAudienceContainer />
        </IntlProviderContainer>
      </Provider>
    )

    wrapper.find(LobbyForAudience).props().transition(ActionTypes.Target.SHOW_BUILD_VILLAGE)
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      type: ActionTypes.Target.SHOW_BUILD_VILLAGE
    })
  })
})
