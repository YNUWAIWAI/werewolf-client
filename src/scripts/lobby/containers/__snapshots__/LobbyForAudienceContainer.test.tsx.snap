// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LobbyForAudienceContainer /> render 1`] = `"<div class=\\"grid\\"><div class=\\"header\\">Lobby for audiences</div><div class=\\"avatar\\"><img class=\\"avatar--image\\" src=\\"\\"><span class=\\"avatar--name\\"></span></div><div class=\\"main-content\\"><div class=\\"village\\"></div></div><div class=\\"aside-content \\"><ul class=\\"compact-menu\\"><li class=\\"compact-menu--item  \\">ID search</li><li class=\\"compact-menu--item  \\">Advanced search</li><li class=\\"compact-menu--item  \\">Refresh</li><li class=\\"compact-menu--item  \\">Return to the main page</li></ul></div></div>"`;
